ermia.cc

```diff
@@ -199,6 +201,37 @@ void ConcurrentMasstreeIndex::GetRecord(transaction *t, rc_t &rc, const varstr &
   }
 }
 
+DirIterator *ConcurrentMasstreeIndex::GetRecordMultiIt(transaction *t, rc_t &rc, const varstr &key) {
+    rc = {RC_INVALID};
+    OID dir_oid = INVALID_OID;
+    std::vector<OID> oids;
+    ermia::varstr tmpval;
+    auto dir_it = new DirIterator(t, table_descriptor);
```

Comment:
`dir_it` should be something maintained (e.g., created on stack on demand) by the caller. Frequent `new` should be avoided here.

Reply:
Yes, now it don't need to be allocated every time, since the OID vector inside has gone

Reply:
Is this the latest up-to-date code? If not we shouldn't review it yet.

Reply:
It is, I will update all changes related to the review later

Reply:
ok, but you mentioned "the OID vector inside has gone?" I assumed you had changes that aren't checked in yet.

Reply:
The OID vector inside class DirIterator is gone so we don't need to allocate space for vector when creating a DirIterator, but some useless code in GetRecordMultiIt such as line 207 should also be removed.

Reply:
Still not sure what you meant: this comment was about L209, not L207.

Reply:
Yes. I mean previously(before this commit) DirIterator has a OID vector inside, but now it is gone. so I can move the dir_it fully to stack now,

Reply:
ok, let's move it to stack then.